src/containers/FormikForm/translateFormHooks.js

```diff
@@ -7,56 +7,27 @@
  */
 
 import { useState } from 'react';
+import dot from 'dot-object';
+import { merge } from 'lodash';
 import { fetchNnTranslation } from '../../modules/translate/translateApi';
 
-export function useTranslateForm(article, setArticle) {
+export function useTranslateApi(element, setElement, fields = {}) {
   const [translating, setTranslating] = useState(false);
 
-  const translateArticle = async () => {
+  const translateFunc = async () => {
     setTranslating(true);
-    const { id, title, metaDescription, introduction, content } = article;
-    const translatedArticleContents = await fetchNnTranslation({
-      id,
-      title,
-      metaDescription,
-      introduction,
-      content,
-    });
-    setArticle({
-      ...article,
-      ...translatedArticleContents.document,
-      language: 'nn',
-    });
-    setTranslating(false);
-  };
-
-  return {
-    translating,
-    translateArticle,
-  };
-}
 
-export function useTranslateConceptForm(concept, setConcept) {
-  const [translating, setTranslating] = useState(false);
-
-  const translateConcept = async () => {
-    setTranslating(true);
-    const { id, title, content } = concept;
-    const translatedConceptContents = await fetchNnTranslation({
-      id,
-      title,
-      content,
-    });
-    setConcept({
-      ...concept,
-      ...translatedConceptContents.document,
+    const translatedContents = await fetchNnTranslation({ ...fields });
+    const document = dot.object(translatedContents.document);
```

Comment:
Oversettertjeneste godtar berre ett nivå med felt i json som sendes inn, så for å støtte nesta objekt brukes dot-object som konverterer f.eks 'podcastMeta.manuscript' til {podcastMeta: {manuscript}}. merge litt lenger nede slår sammen originalt objekt og det oversatte objektet til et samla objekt uten at podcastMeta fra det oversatte objektet overskriver fra originalt.

Reply:
kanskje legge til en kommentar på det over?

Reply:
Ja, tenkte tanken.